### PR TITLE
feat: [STO-6008]: Remove account requirement for STO token

### PR DESCRIPTION
**Pull request by ahokinson**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48618)

### CODE REVIEW

**Review by GabeAboy — DISMISSED** [review]

**Review by chris-harness — APPROVED** [review]

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskUtils.java [hunk 1]
```diff
@@ -529,6 +530,7 @@ public Map<String, String> getSTOServiceEnvVariables(String accountId) {
 
     envVars.put(STO_SERVICE_TOKEN_VARIABLE, stoServiceToken);
     envVars.put(STO_SERVICE_ENDPOINT_VARIABLE, stoServiceBaseUrl);
+    envVars.put(STO_INTERNAL_URL_VARIABLE, stoServiceInternalUrl);
```
Comment (r0610205) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskUtils.java:533]: We don't need to pass it through

File: 323-sto-utilities/src/main/java/io/harness/common/STOExecutionConstants.java [hunk 2]
```diff
@@ -14,5 +14,6 @@
 public class STOExecutionConstants {
   // These are environment variables to be set on the pod for talking to the STO service.
   public static final String STO_SERVICE_ENDPOINT_VARIABLE = "HARNESS_STO_SERVICE_ENDPOINT";
+  public static final String STO_INTERNAL_URL_VARIABLE = "HARNESS_STO_INTERNAL_URL";
```
Comment (r0610205) [inline 323-sto-utilities/src/main/java/io/harness/common/STOExecutionConstants.java:17]: there is no need for this constant

File: 315-sto-manager/app/src/main/java/io/harness/app/telemetry/STOTelemetryPublisher.java [hunk 3]
```diff
@@ -35,29 +35,26 @@ public class STOTelemetryPublisher {
...
-          log.info("Scheduled STOTelemetryPublisher event sent! for account {}", usage.accountId);
-        }
+        HashMap<String, Object> map = new HashMap<>();
+        map.put(GROUP_TYPE, ACCOUNT);
+        map.put(GROUP_ID, usage.accountId);
+        map.put(ACCOUNT_DEPLOY_TYPE, System.getenv().get(DEPLOY_VERSION));
+        map.put(LICENSE_USAGE, max(ceil(usage.scanCount / 100.0), usage.developerCount));
+        telemetryReporter.sendGroupEvent(usage.accountId, null, map, Collections.singletonMap(ALL, true),
+            TelemetryOption.builder().sendForCommunity(false).build());
+        log.info("Scheduled STOTelemetryPublisher event sent! for account {}", usage.accountId);
```
Comment (r0610205) [inline 315-sto-manager/app/src/main/java/io/harness/app/telemetry/STOTelemetryPublisher.java:57]: let's remove exclamation mark

File: 332-ci-manager/chart/templates/config.yaml [hunk 4]
```diff
@@ -48,6 +48,7 @@ data:
   TI_SERVICE_INTERNAL_URL: 'http://ti-service:8078/'
   API_URL: '{{ .Values.global.loadbalancerURL }}/ng/#/'
   STO_SERVICE_ENDPOINT: '{{ .Values.global.loadbalancerURL }}/sto/'
+  STO_INTERNAL_URL: '{{ .Values.global.stoInternalURL }}'
```
Comment (r0610205) [inline 332-ci-manager/chart/templates/config.yaml:51]: there is no such value

Comment (r0610205) [reply 332-ci-manager/chart/templates/config.yaml:51]: Should be   STO_SERVICE_INTERNAL_ENDPOINT: 'http://sto-core.{{ .Release.Namespace }}.svc.cluster.local:4000/'

**Review by r0610205 — APPROVED** [review]

File: 315-sto-manager/chart/templates/config.yaml [hunk 5]
```diff
@@ -18,6 +18,7 @@ data:
   TI_SERVICE_ENDPOINT: '{{ .Values.global.loadbalancerURL }}/ti-service/'
   TI_SERVICE_INTERNAL_ENDPOINT: 'http://ti-service.{{ .Release.Namespace }}.svc.cluster.local:8078/'
   STO_SERVICE_ENDPOINT:  '{{ .Values.global.loadbalancerURL }}/sto/'
+  STO_INTERNAL_URL: '{{ .Values.global.stoInternalURL }}'
```
Comment (r0610205) [inline 315-sto-manager/chart/templates/config.yaml:21]: This one should be removed

**Review by r0610205 — APPROVED** [review]

**Comment by ahokinson** [comment]
trigger uts

**Comment by ahokinson** [comment]
trigger uts

**Comment by ahokinson** [comment]
trigger uts

**Review by jamie-harness — APPROVED** [review]

**Review by jamie-harness — APPROVED** [review]

**Review by r0610205 — APPROVED** [review]

**Pull request merged**